README.md

```diff
@@ -297,6 +298,78 @@ When building with CMake, submodules are also recursively cloned automatically.
 can be passed as a CMake flag to disable this functionality. This is useful when you'd like to build CMake while using a
 different commit from a submodule.
 
+### Installation
+
+The C SDK libraries and platform abstractions can be installed to a file system
+through CMake.
+```sh
+cmake .. -DBUILD_DEMOS=0 -DBUILD_TESTS=0
+sudo make install
+```
+Note that because `make install` will automatically build the `all` target, it may
+be useful to disable building demos and tests with `-DBUILD_DEMOS=0 -DBUILD_TESTS=0`
+unless they have already been configured. Super-user permissions may be needed if
+installing to a system include or system library path.
+
+To install only a subset of all libraries, pass `-DINSTALL_LIBS` to install only
+the libraries you need. By default, all libraries will be installed, but you may
+exclude any library that you don't need from this list:
+```
+-DINSTALL_LIBS="DEFENDER;SHADOW;JOBS;OTA;OTA_HTTP;OTA_MQTT;BACKOFF_ALGORITHM;HTTP;JSON;MQTT;PKCS"
+```
+
+By default, the location of headers will be in the default system path for headers
+such as `/usr/local/include` for Linux. For libraries, this will likely
+be `/usr/local/lib` for Linux.
+
+Upon entering `make install`, the location of each library will be specified first
+followed by the location of all installed headers:
+```
+-- Installing: /usr/local/lib/libaws_iot_defender.so
+-- Installing: /usr/local/lib/libaws_iot_shadow.so
+...
+-- Installing: /usr/local/include/aws/defender.h
+-- Installing: /usr/local/include/aws/defender_config_defaults.h
+-- Installing: /usr/local/include/aws/shadow.h
+-- Installing: /usr/local/include/aws/shadow_config_defaults.h
+```
+The libraries can now be linked to any build system of your choice (e.g. CMake):
+```cmake
+target_link_libraries(execName PUBLIC aws_iot_defender aws_iot_shadow)
+```
+Furthermore, the headers can now be included to any compilation unit as system headers:
+```c
+#include <aws/defender.h>
+#include <aws/shadow.h>
+```
+
+Note that by default, CMake will install to the default system path for headers and libraries of your OS.
+However, an installation path of your choice can be specified by passing the
+following flags through CMake:
+```sh
+cmake .. -DBUILD_DEMOS=0 -DBUILD_TESTS=0 \
+-DCSDK_HEADER_INSTALL_PATH="header-path" -DCSDK_LIB_INSTALL_PATH="lib-path"
+sudo make install
+```
+
+POSIX platform abstractions are used together with the C-SDK libraries in the demos.
+By default, these abstractions are also installed but can be excluded by passing
+the flag: `-DINSTALL_PLATFORM_ABSTRACTIONS=0`.
```

Comment:
Shouldn't the installation of the platform libraries be disabled by default to prevent libraries not specified in the `INSTALL_LIBS` variable from being installed to the system?

Reply:
Because everything is installed by default unless otherwise specified, I decided to also enable this option by default. Also, I feel like `-DINSTALL_LIBS` should not affect `-DINSTALL_PLATFORM_ABSTRACTIONS` the same way `-DBUILD_DEMOS` does not affect `-DBUILD_TESTS`.